Source/Scene/Scene.js

```diff
@@ -3549,6 +3543,279 @@ define([
+                    position = Ray.getPoint(ray, distance);
+                    break;
+                }
+            }
+
+            if (defined(position) && (scene.mode !== SceneMode.SCENE3D)) {
+                Cartesian3.fromElements(position.y, position.z, position.x, position);
+
+                var projection = scene.mapProjection;
+                var ellipsoid = projection.ellipsoid;
+
+                var cartographic = projection.unproject(position, scratchPickPositionCartographic);
+                ellipsoid.cartographicToCartesian(cartographic, position);
+            }
+        }
+
+        scene._view = scene._defaultView;
+        context.endFrame();
+        callAfterRenderFunctions(scene);
+
+        if (!defined(object) && !defined(position)) {
+            return;
+        }
+
+        return {
+            object : object,
+            position : position
+        };
+    }
+
+    /**
+     * Returns an object with a `primitive` property that contains the first (top) primitive in the scene
+     * hit by the ray or undefined if nothing is hit. Other properties may potentially be set depending on the type of primitive.
+     * <p>
+     * When a feature of a 3D Tiles tileset is picked, <code>pick</code> returns a {@link Cesium3DTileFeature} object.
+     * </p>
+     *
+     * @private
+     *
+     * @param {Ray} ray The ray to pick from.
+     * @returns {Object} Object containing the picked primitive.
+     *
+     * @see Scene#pick
+     */
+    Scene.prototype.pickFromRay = function(ray) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('ray', ray);
+        //>>includeEnd('debug');
+        var pickResult = pickFromRay(this, ray, false, true);
+        if (defined(pickResult)) {
+            return pickResult.object;
+        }
+    };
+
+    /**
+     * Returns the cartesian position of the first intersection of the ray or undefined if nothing is hit.
+     *
+     * @private
+     *
+     * @param {Ray} ray The ray to pick from.
+     * @param {Cartesian3} [result] The object on which to restore the result.
+     * @returns {Cartesian3} The cartesian position.
+     *
+     * @exception {DeveloperError} Picking from the depth buffer is not supported. Check pickPositionSupported.
+     */
+    Scene.prototype.pickPositionFromRay = function(ray, result) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('ray', ray);
+        if (!this.pickPositionSupported) {
+            throw new DeveloperError('Picking from the depth buffer is not supported. Check pickPositionSupported.');
+        }
+        //>>includeEnd('debug');
+        var pickResult = pickFromRay(this, ray, true, false);
+        if (defined(pickResult)) {
+            return Cartesian3.clone(pickResult.position, result);
+        }
+    };
+
+    /**
+     * Returns a list of objects, each containing a `primitive` property, for all primitives hit
```

Comment:
What other properties does each object have?  Why not just return an array of primitives?

Reply:
OK, can this and the other functions provide a bit more context?